docs/src/man/iterative_refinement.md

```diff
@@ -11,61 +12,213 @@ Depth = 3
+```
 
-```@example ir
-import LazySets.Approximations:Approximation2D, refine
+Hence, we can make again a refinement of that approximation.
 
-const DIR_EAST, DIR_NORTH, DIR_WEST, DIR_SOUTH = [1., 0.], [0., 1.], [-1., 0.], [0., -1.]
+```@example example_iterative_refinement
+(r1, r2) = refine(Ω, 1)
+Ω.approx_list[1] = r1
+insert!(Ω.approx_list, 2, r2)
 
+plot(b, 1e-3, aspectratio=1, alpha=0.3)
+plot!(Approximations.tohrep(Ω), alpha=0.2, color="orange")
 ```
+
+The criterion for an approximation being refinable is that we can properly define
+a normal direction `ndir`. This boils down to checking for the following
+"degenerate" cases:
+
+1. ``p_1`` and ``p_2`` overlap.
+2. ``p_1`` and ``q`` overlap.
+1. ``p_2`` and ``q`` overlap.
+
+Moreover, we include the condition `approx_error > TOL` where `TOL` is the floating
+point epsilon in the given numerical precision.
+
+## Algorithm
+
+Having presented the individual steps, we give the pseudocode of the iterative
+refinement algorithm, see `approximate(S, ε)`.
+
+The algorithm consists of the following steps:
+
+1. *Initialization*. The approximation is initialized with box directions,
+   i.e. it starts with four `LocalApproximation` objects. Let `i=1`.
+2. *Refinement loop*. If the local approximation at index `i` has an error greater
+   than the threshold `ε`, then refine. Otherwise, increment `i <- i+1`.
+3. *Redundancy check*. Insert the refined right approximation at position `i`,
+   and check whether the left approximation is redundant or not with respect to the one
+   at position `i+1`. Checking for redundancy amounts to checking for overlap of both
+   `p1` and `q`. Then, either substitute at `i+1` or insert (keeping the approximation
+    at `i+1`) depending on the redundancy check.
+4. *Stopping criteria*. If the index `i` is smaller or equal than the current length of
+   the approximations list, go again to step 2.
+
+Observe that the algorithm finishes when all approximations are such that
+their associated error is smaller than `ε`, hence the Hausdorff distance between
+`S` and its polygonal overapproximation is no greater than `ε`.
+
+## Example
+
+As a final example consider the iterative refinement of the ball `b` for different
+values of the approximation threshold `ε`.
+
+```@example example_iterative_refinement
+import LazySets.Approximations.overapproximate
+
+p0 = plot(b, 1e-6, aspectratio=1)
+p1 = plot!(p0, overapproximate(b, 1.), alpha=0.4, aspectratio=1)
+
+p0 = plot(b, 1e-6, aspectratio=1)
+p2 = plot!(p0, overapproximate(b, 0.1), alpha=0.4, aspectratio=1)
+
+p0 = plot(b, 1e-6, aspectratio=1)
+p3 = plot!(p0, overapproximate(b, 0.01), alpha=0.4, aspectratio=1)
+
+plot(p1, p2, p3, layout=(1, 3))
+```
+
+We can check that the error is getting smaller with `ε` in each case:
+
+```@example example_iterative_refinement
+f = x -> (minimum(x), maximum(x))
+g = ε ->  f([ai.err for ai in Approximations.approximate(b, ε).approx_list])
```

Comment:
Maybe add `import LazySets.Approximations.approximate` to remove `Approximations.approximate` twice?

Reply:
yes